### PR TITLE
Add aiccl support

### PR DESCRIPTION
**Pull request by maaquib**
## Description ##

Adds aiccl backend support

### TODO
- [x] Add ci test. We don't have `p4dn` runners right now
- [x] Add `aiccl` to docker conainer

### CODE REVIEW

File: engines/python/setup/djl_python/rolling_batch/lmi_dist_rolling_batch.py [hunk 1]
```diff
@@ -58,6 +58,8 @@ def _init_model(self, kwargs, model_id_or_path):
         revision = self.properties.get('revision', None)
         paged_attention = self.properties.get("paged_attention",
                                               "true").lower() == "true"
+        use_aiccl_backend = self.properties.get("use_aiccl_backend",
```
Comment (lanking520) [inline engines/python/setup/djl_python/rolling_batch/lmi_dist_rolling_batch.py:61]: let's hide this to deeper

File: tests/integration/llm/prepare.py [hunk 2]
```diff
@@ -509,6 +509,18 @@
     },
 }
 
+lmi_dist_aiccl_model_list = {
```
Comment (lanking520) [inline tests/integration/llm/prepare.py:512]: we will need P4D to test this

Comment (maaquib) [reply tests/integration/llm/prepare.py:512]: Yes. I added this as a todo in the PR description for tracking

**Review by lanking520 — APPROVED** [review]

File: .github/workflows/llm_integration.yml [hunk 3]
```diff
@@ -8,7 +8,7 @@ on:
         required: false
         default: ''
       run_test:
-        description: 'Run only the tests you need [ft, ds, hf, aot, lora-correctness]'
+        description: 'Run only the tests you need [ft, ds, hf, aot, lora-correctness, smoothquant]'
```
Comment (lanking520) [inline .github/workflows/llm_integration.yml:11]: what is this for? Do we expect it to work with smoothquant?

File: serving/docker/deepspeed.Dockerfile [hunk 4]
```diff
@@ -51,6 +52,7 @@ ENV HUGGINGFACE_HUB_CACHE=/tmp/.cache/huggingface/hub
 ENV TRANSFORMERS_CACHE=/tmp/.cache/huggingface/transformers
 ENV PYTORCH_KERNEL_CACHE_PATH=/tmp/.cache
 ENV BITSANDBYTES_NOWELCOME=1
+ENV USE_AICCL_BACKEND=True
```
Comment (lanking520) [inline serving/docker/deepspeed.Dockerfile:55]: set to true or TRUE

File: .github/workflows/llm_integration.yml (hunk 3 above)
Comment (maaquib) [reply .github/workflows/llm_integration.yml]: Removed

File: serving/docker/deepspeed.Dockerfile (hunk 4 above)
Comment (maaquib) [reply serving/docker/deepspeed.Dockerfile]: Fixed

**Review by lanking520 — APPROVED** [review]

**Pull request merged**